test/integration/patientMatches.js

```diff
@@ -0,0 +1,258 @@
+/* global expect, agent */
+
+// eslint-disable no-unused-expressions
+
+const UUID = require('uuid').v4;
+
+const helpers = require('./helpers');
+
+const db = require('../../server/lib/db');
+
+function uuid() {
+  return UUID().toUpperCase().replace(/-/g, '');
+}
+
+// Reuse the same location/project uuids for all mock patients
+const groupUuid = '4DE0FE47177F4D30B95FCFF8166400B4';
+const locationUuid = '1F162A109F6747889EFFC1FEA42FCC9B';
+const projectId = 1;
+
+function addDebtorSQL(debtorUuid, text) {
+  return `INSERT INTO debtor (uuid, group_uuid, text) VALUES (0x${debtorUuid}, 0x${groupUuid}, '${text}');`;
+}
+
+function addPatientSQL(pinfo) {
+  return 'INSERT INTO patient ('
+   + 'uuid, project_id, debtor_uuid, display_name, sex, dob, dob_unknown_date, '
+   + 'origin_location_id, current_location_id, user_id) '
+   + `VALUES (0x${pinfo[0]}, ${projectId}, 0x${pinfo[5]},'${pinfo[1]}', '${pinfo[2]}', `
+   + `'${pinfo[3]}', ${pinfo[4]}, 0x${locationUuid}, 0x${locationUuid}, 1`
+   + ');';
+}
+
+const mockPatients = [
+  // uuid, display_name, sex, dob, dob_unknown_date, debtorUuid
+  [uuid(), 'John Smith', 'M', '1993-08-01', 'FALSE', uuid()],
+  [uuid(), 'Jon Smith', 'M', '1995-01-01', 'TRUE', uuid()],
+
+  [uuid(), 'John Jones Mitchum', 'M', '1980-01-01', 'TRUE', uuid()],
+  [uuid(), 'John Janes Mitchum', 'M', '1981-04-21', 'FALSE', uuid()],
+  [uuid(), 'John Jenes Matchim', 'M', '1984-01-01', 'TRUE', uuid()],
+
+  [uuid(), 'Lynn H. Black', 'M', '1970-08-01', 'FALSE', uuid()],
+  [uuid(), 'Lynn Black', 'F', '1981-01-01', 'TRUE', uuid()],
+];
+
+let initialNumPatients = null;
+
+describe('(/patients) Find matching patients', () => {
+
+  // Note how many patients we had to start with
+  it('Save starting number of patients', () => {
+    return agent.get('/patients')
+      .then((res) => {
+        initialNumPatients = res.body.length;
+      })
+      .catch(helpers.handler);
+  });
+
+  // Add all the mock patients for this test
+  mockPatients.forEach(p => {
```

Comment:
I do not believe that async works properly in Array forEach statements.  Here is what I would do:

```js
const promises = await Promise.all(mockPatients.map(p => {
  return Promise.all([db.exec(), db.exec()]);
}));

// Now do your asserts
const isAllTrue = promises.every(val => !!val === true);
expect(isAllTrue).to.equal(true);
```

I wrote this on a smartphone, apologies for syntax errors :)

Reply:
I've tried multiple ways to fix this.   I cannot force the setup DB queries to reliably execute before running the tests that depend on them.  Tried about 10 different things without luck -- including a global before() function.  Still either run into SQL queries not completing early enough or DB LOCK problems (running multiple queries at the same time).  Look forward to chatting with you about this.

Reply:
With help from @jniles, I implemented this fix.